core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/handlers/MigrationPropertyTester.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2021 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.data.migration.handlers;
+
+import java.util.Arrays;
+
+import org.eclipse.core.expressions.PropertyTester;
+
+/**
+ * 
+ */
+public class MigrationPropertyTester extends PropertyTester {
+
+  @Override
+  public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+    // check if is project, aird or model migration
+    switch(property) {
+    case "isValidMigrationProject":
+      return new ProjectMigrationHandler().isValidSelection(Arrays.asList(receiver));
```

Comment:
When we disable commands through "Window / Preferences / Capella / CommandConfigurability / Migration"
the handler on the command is replaced by another one by "PreferenceHandler.getInstance(command).activateHandler

If we use the "classical one" here, the command will be disabled and still visible even if we used a visibleWhen in the menu command.
In the new patchset, we retrieve the current command handler (classical one or PreferenceHelper disabling one) then look if it is enabled.